structure_tests/structure_test_v1.go

```diff
@@ -107,3 +120,45 @@ func (st StructureTestv1) RunFileContentTests(t *testing.T) {
 		}
 	}
 }
+
+// Given a list of environment variables and a list of lists of strings,
+// retrieve each environment variable's value and replace all occurrences
+// of it in each list of strings provided.
+func SubstituteEnvVars(t *testing.T, vars []string, lists ...*[]string) {
+	for _, env_var := range vars {
+		value := os.Getenv(env_var)
+		if value == "" {
+			t.Fatalf("Variable %s not found in environment!", env_var)
+		}
+		for _, list := range lists {
+			for i := range *list {
+				str := (*list)[i]
+				(*list)[i] = strings.Replace(str, "$"+env_var, value, -1)
+			}
+		}
+	}
+}
+
+func ProcessCommands(t *testing.T, commands []string) {
```

Comment:
yeah, this is a pretty bad name. this is just a helper method to run the setup and teardown commands.

i struggled with this a bit when figuring out how to define it in the test json. it's a bit awkward to me to have the `setup` and `teardown` fields in the JSON themselves also have nested `command` and `flags` fields. so instead i just had them defined as individual strings, so it would look like
```
"setup": ["install_node v5.9.0"]
```
as opposed to 
```
"setup": [
  {
    "command": "install_node",
    "flags": ["v5.9.0"]
  }
]
```

maybe it makes more sense to do it the second way though?